### PR TITLE
ReDelay.cpp

### PR DESCRIPTION
**Pull request by alexvann**
Added a check to avoid dereferencing null pointer.

### CODE REVIEW

File: modules/ReDelay/ReDelay.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ void handleInput(SokuLib::KeymapManager *base) {
 
 	auto &mgr = SokuLib::getBattleMgr();
 
-	if (base == mgr.leftCharacterManager.keyManager->keymapManager) {
```
Comment (Gegel85) [inline modules/ReDelay/ReDelay.cpp:72]: Does it even work though ?
You are checking for mgr.**right**CharacterManager.keyManager but in the condition it dereferences mgr.**left**CharacterManager.keyManager.

File: modules/ReDelay/ReDelay.cpp [hunk 2]
```diff
@@ -69,7 +69,7 @@ void handleInput(SokuLib::KeymapManager *base) {
...
+	if (mgr.rightCharacterManager.keyManager && base == mgr.leftCharacterManager.keyManager->keymapManager) {
```
Comment (Gegel85) [inline modules/ReDelay/ReDelay.cpp:72]: I commented on the wrong side

Comment (alexvann) [reply modules/ReDelay/ReDelay.cpp:72]: Yeah `left` ofc
That's my style %)

**Comment by delthas** [comment]
Does this solve an actual bug or is this just bad linting?

**Comment by alexvann** [comment]
> Does this solve an actual bug or is this just bad linting?
Definitely solves. The issue is that for the game to consider the player as a dummy `keyManager` must be zeroed. I set P1 as dummy and then ReDelay breaks down.
This check exists in whole the game code.

**Comment by delthas** [comment]
Please edit your commit with the right check first.

**Comment by delthas** [comment]
Then up to you @Gegel85 for whether to merge this.

**Review by Gegel85 — APPROVED** [review]

**Pull request merged**